monitor/monitor.c

```diff
@@ -205,6 +207,14 @@ static pid_t fork_exec(int i)
                 args[a ++] = "--tag";
                 args[a ++] = tag;
             }
+            if ((period = getenv(ENV_HEALTH_CHECK_PERIOD)) != NULL) {
+                args[a ++] = "--period";
+                args[a ++] = period;
+            } 
+            if ((retry_max = getenv(ENV_RETRY)) != NULL) {
+                args[a ++] = "--retry_max";
+                args[a ++] = retry_max;
+            }            
```

Comment:
HEALTH_CHECK_PERIOD/MAX_RETRY can be configured in scanner yaml
monitor as the entry process in scanner container, it parses the env vars & translate them to --period/--retry_max arguments when creating scanner process

Reply:
@pohanhuangtw

Reply:
Hi @holyspectral,
The scanner.go is triggered by monitor.c. To configure the HEALTH_CHECK_PERIOD and MAX_RETRY correctly, we need to add these values to the environment variables. This way, the scanner can read them correctly.

Reply:
Hi @williamlin-suse,

After a quick discussion with @holyspectral, we realized that monitor.c may be shared by many components, and we're not sure if making changes might break some behavior. Would it be okay to use os.GetEnv() to read the environment variables instead of passing them through monitor.c?

Reply:
To add, I think monitor's design purpose is to monitor and restart its child process, so it's not a perfect place to write component-specific logic.  Technically we can still put some logic into monitor if we really want to, but maybe we should only keep generic ones there.

Reply:
scanner repo has its own monitor code.
what do you mean "monitor.c may be shared by many components" ?
I think another PR is needed for neuvector-helm repo (github.com/neuvector/neuvector-helm/charts/core/templates/scanner-deployment.yaml) regarding the 2 new env variables added in this PR.

Reply:
I think env vars can be provided to scanner pod via the existing `cve.scanner.env` already, but we probably should document these settings somewhere for sure.  Do you have any places in mind?

Reply:
We can submit a jira case to Nuno for doc update

Reply:
I do not think we should export those variables unless they are useful for customers to escalate their situation. @pohanhuangtw please use a default pair of the reasonable value instead.